src/algorithms/union_find.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <vector>
+
+struct UnionFind {
+	int new_var() {
+		int result = repr.size();
+		repr.push_back(result);
+		return result;
+	}
```

Comment:
The concept "new var" is related to the use cases of the union find, maybe a more generic name would be better, because we are trying to abstract the algorithm away from the codebase. My suggestions would be to keep "new" or maybe use "create" or "add". As for "var", maybe use "node".

Reply:
yeah, good point. so `new_node`? Coming right up

I would like to start using the word `create` more often, as it evokes the picture that a value is being produced, but maybe some side effects happen as well.